src/map/storage.c

```diff
@@ -329,22 +348,26 @@ int storage_storageaddfromcart(struct map_session_data* sd, int index, int amoun
 int storage_storagegettocart(struct map_session_data* sd, int index, int amount)
 {
 	int flag = 0;
+	struct item *it = NULL;
+
 	nullpo_ret(sd);
 
-	if( index < 0 || index >= MAX_STORAGE )
+	if (index < 0 || index >= MAX_STORAGE)
 		return 0;
 
-	if( sd->status.storage.items[index].nameid <= 0 )
+	it = &VECTOR_INDEX(sd->storage, index);
+
+	if (it->nameid <= 0)
 		return 0; //Nothing there.
 
-	if( amount < 1 || amount > sd->status.storage.items[index].amount )
+	if (amount < 1 || amount > it->amount)
 		return 0;
 
-	if( (flag = pc->cart_additem(sd,&sd->status.storage.items[index],amount,LOG_TYPE_STORAGE)) == 0 )
-		storage->delitem(sd,index,amount);
+	if ((flag = pc->cart_additem(sd, it, amount, LOG_TYPE_STORAGE)) == 0)
+		storage->delitem(sd, index, amount);
 	else {
 		clif->dropitem(sd, index,0);
-		clif->cart_additem_ack(sd,flag == 1?0x0:0x1);
+		clif->cart_additem_ack(sd, flag == 1?0x0:0x1);
```

Comment:
Code-styling will be handled in a separate commit, once the main commit has been reviewed. But thanks for your findings.